jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/Resources.kt

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation
+
+interface Function : Resource {
+    fun codeLocation(): String
+    fun setCodeLocation(location: String)
+    fun runtime(): String
+    fun handler(): String
+}
+
+const val LAMBDA_FUNCTION_TYPE = "AWS::Lambda::Function"
+class LambdaFunction(private val delegate: Resource) : Resource by delegate, Function {
+    override fun setCodeLocation(location: String) {
+        setScalarProperty("Code", location)
+    }
+
+    override fun codeLocation(): String = getScalarProperty("Code")
+
+    override fun runtime(): String = getScalarProperty("Runtime")
+
+    override fun handler(): String = getScalarProperty("Handler")
+
+    override fun toString(): String = logicalName
```

Comment:
nit: reckon we could pull this up to `Resource`

Reply:
Can't since it is defined in Any :(

Reply:
boo :(